### DISCUSSION TITLE
Beginning to contribute in open source!

### DISCUSSION THREAD

**Discussion by ItsPiyusha** [EddieHubCommunity/support#6330] [First-Timers] [answered]
How to understand already existing code in open source project? What are fundamentals of repository?

**Comment by ItsPiyusha** [comment]
Here is a [source](https://medium.com/code-factory-berlin/github-repository-structure-best-practices-248e6effc405) I found on Google.

**Comment by NaNshekhar04** [comment, marked as answer]
@ItsPiyusha 
>This guide is from 2020 MLH Program which will help you explore & understand large codebases !
https://pncnmnp.github.io/blogs/oss-guide.html

>Also this is a video link which will help you practically !

https://www.youtube.com/watch?v=kA1T0zDPtQo

**Reply by ItsPiyusha** [reply]
thanks, I saw the video, so basically the video is how to understand large codebases, and the answer is we don't have to. I'll try that.

**Reply by dipankr** [reply]
good tldr 😁👍

**Comment by samurai-techlead** [comment]
@ItsPiyusha 
It's really helpful

**Comment by priyankarpal** [comment]
@ItsPiyusha 

**To understand existing code in an open-source project, you can start by reading the documentation and comments in the code. Understanding the project's file and directory structure, as well as the overall architecture, can also be helpful. Additionally, you can try running the code and experimenting with it to see how it works.**

**As for the fundamentals of a repository, it is a location where files and directories are stored and managed, typically in the context of version control. In an open-source project, the repository is typically hosted on a website such as GitHub, where other developers can access and contribute to the code. The repository usually contains the entire history of the project, including all the changes and contributions made by different developers.**

**You can also look into the README.md of the repository which will give you an overview of the project, what it does, the dependencies, and how to set it up and run it.**

**Comment by pritamrajput** [comment]
How to understand an issue ? I'm completer beginner. I tried understanding it but didn't succeed. Can I get any help ?

**Reply by mikeysan** [reply]
Perhaps a more direct question would offer folks the chance to help you. An open-ended question in tech is not always the best way to go.

What are you trying to do? 
What have you done about it so far?
Where are you stuck?

Show that you have done some research or attempted something and folks are more likely to help you get through the next step.

For example: You said "how to understand an issue?"

What issue?
What about the issue do you not understand?
Is the issue marked as a beginner or first-time issue?

**Comment by priyankarpal** [comment]
@pritamrajput if you want to contribute LinkFree then please check this official doc. Here you will find everything https://linkfree.eddiehub.io/docs/quickstart

**Reply by pritamrajput** [reply]
Thank you @priyankarpal for the reply. I have contributed to linkfree and made my linkfree profile

**Comment by Arsaluddin** [comment]
I can't understand where to start reading the code 
any suggestions please

**Reply by Panquesito7** [reply]
This is a very very good question. For bigger projects, I suggest you take a look at the documentation and see how the project works exactly so that you get familiarized with the code to contribute later on.

For smaller projects, you can usually take a look at the code on multiple files to get familiarized with the project and it shouldn't take long (it depends on your knowledge of programming as well as the complexity of the project, of course).

I hope this helped. Let us know if you need any more help. 🙂

**Comment by marrious11** [comment]
Oh! Thank you all for this wonderful resource. It just came at the right time for me.

**Comment by adisuyash** [comment]
I also have one question that I started to contribute with to documentations because I am unable to find the relevant repos which I can contribute to as a Developer as I strongly wish to do so.
My Tech Stack is not so much as I have just HTML CSS & Markdown(for docs) and I have started to learn JavaScript as well.
My question is how can find good-first-issues for HTML CSS because some repos have good-first-issue labels but I can't understand what's going on?

**Reply by priyankarpal** [reply]
You can find open-source projects on GitHub, social media etc. Join communities, use hashtags to find projects also check these websites to see lots of projects 
- https://finder.eddiehub.io/
- https://www.oswc.is/
and I also created a list of projects you can check 
- https://github.com/stars/priyankarpal/lists/open-source-projects

**Reply by adisuyash** [reply]
Thanks alot ❤️
